src/System.Security.Cryptography.Pkcs/tests/CmsSignerTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Xunit;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Security.Cryptography.Pkcs.Tests
+{
+    public class CmsSignerTests
+    {
+        [Fact]
+        public void SignCmsUsingAsymmetricSignatureFormatter()
+        {
+            byte[] contentBytes = { 9, 8, 7, 6, 5 };
+
+            byte[] signedData = GetSignedData(contentBytes, out byte[] certBytes);
+            Assert.True(ByteArrayContains(signedData, contentBytes));
+
+            using (X509Certificate2 cert = new X509Certificate2(certBytes))
+            {
+                Assert.Null(cert.PrivateKey);
```

Comment:
`Assert.False(cert.HasPrivateKey)`.  While this assert will also happen to be true it would be misleadingly-throwy in degenerate failure cases.

Reply:
makes sense, didn't know about this property